components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -50,6 +53,14 @@ absl::optional<std::string> CreateWireguardConfig(
   return config;
 }
 
+wireguard::WireguardKeyPair GenerateNewX25519Keypair() {
+  crypto::EnsureOpenSSLInit();
+  uint8_t pubkey[32] = {}, privkey[32] = {};
+  X25519_keypair(pubkey, privkey);
+  return std::make_tuple(base::Base64Encode(pubkey),
+                         base::Base64Encode(privkey));
```

Comment:
This method of encoding does not operate in constant time. From what I can tell this isn't going to be attackable externally, but after digging deeper into the API design I was a bit curious why we need these keys to be encoded. Would it be possible to just pass these keys around as `uint8_t` only or is some aspect of wireguard requiring this further down the stack that I haven't spotted yet?

Reply:
we actually do not need binary representation, we have to use base64 versions inside constructed config file here https://github.com/brave/brave-core/blob/master/components/brave_vpn/common/wireguard/wireguard_utils.cc#L19.

Reply:
In that case for a defense in depth approach we should be making sure that the encoding is being done in constant time. I wouldn't say we need to block on it though because from what I can tell the threat of someone sidechannel attacking this is near negligible.

Reply:
done